Runtime/Tracking/Velocity/ArtificialVelocityApplier.cs

```diff
@@ -0,0 +1,131 @@
+        /// </summary>
+        [Serialized, Validated]
+        [field: DocumentedByXml]
+        public float AngularDrag { get; set; } = 0.5f;
+
+        /// <summary>
+        /// The tolerance the velocity can be within zero to be considered nil.
+        /// </summary>
+        [Serialized, Validated]
+        [field: DocumentedByXml]
+        public float NilVelocityTolerance { get; set; } = 0.001f;
+
+        /// <summary>
+        /// The tolerance the angular velocity can be within zero to be considered nil.
+        /// </summary>
+        [Serialized, Validated]
+        [field: DocumentedByXml]
+        public float NilAngularVelocityTolerance { get; set; } = 0.001f;
+
+        /// <summary>
+        /// The routine to handle the deceleration to the object based on the drag over time.
+        /// </summary>
+        private Coroutine deceleration;
+
+        /// <summary>
+        /// Applies the velocity data to the <see cref="target"/>.
+        /// </summary>
+        public virtual void Apply()
+        {
+            CancelDeceleration();
+            deceleration = StartCoroutine(BeginDeceleration());
+        }
+
+        /// <summary>
+        /// Reset the velocity to zero.
+        /// </summary>
+        public virtual void ResetVelocity()
+        {
+            Velocity = Vector3.zero;
+        }
+
+        /// <summary>
+        /// Reset the angular velocity to zero.
+        /// </summary>
+        public virtual void ResetAngularVelocity()
+        {
+            AngularVelocity = Vector3.zero;
+        }
+
+        protected virtual void OnDisable()
+        {
+            CancelDeceleration();
+        }
+
+        /// <summary>
+        /// Begins decelerating the <see cref="target"/> based on any opposing drag forces.
+        /// </summary>
+        /// <returns>An Enumerator to manage the running of the Coroutine.</returns>
+        protected virtual IEnumerator BeginDeceleration()
+        {
+            while (!Velocity.ApproxEquals(Vector3.zero, NilVelocityTolerance) || !AngularVelocity.ApproxEquals(Vector3.zero, NilAngularVelocityTolerance))
+            {
+                Velocity = Vector3.Slerp(Velocity, Vector3.zero, Drag * Time.deltaTime);
+                AngularVelocity = Vector3.Slerp(AngularVelocity, Vector3.zero, AngularDrag * Time.deltaTime);
+                Target.transform.localRotation *= Quaternion.Euler(AngularVelocity);
+                Target.transform.localPosition += Velocity * Time.deltaTime;
+                yield return null;
+            }
+
+            deceleration = null;
+            Velocity = Vector3.zero;
+            AngularVelocity = Vector3.zero;
+        }
+
+        /// <summary>
+        /// Cancels the deceleration routine.
+        /// </summary>
+        protected virtual void CancelDeceleration()
+        {
+            if (deceleration != null)
+            {
+                StopCoroutine(deceleration);
+            }
+            deceleration = null;
```

Comment:
This is a "wasteful" instruction if `deceleration` was null in the above `if` and one reads it and wonders if there's any magic happening in a potential setter or something when this already was `null`.

Long story short: **Negate** the `if` and change its body to just do an early `return`. Then the `StopCouroutine` call and the set to `null` can be done afterwards, outside the `if`.

Reply:
tldr; this?
```
if (decelerationRoutine == null)
{
    return;
}

StopCoroutine(decelerationRoutine);
decelerationRoutine = null;
```

Reply:
👍